service/control.go

```diff
@@ -69,10 +76,36 @@ func (ctrl *Control) postWorkflow(c *gin.Context) {
 		Owner:       body.Owner,
 		IsActive:    body.IsActive,
 	}
-	ctrl.db.Create(&wf)
+	// upsert workflow
+	ctrl.db.Clauses(
+		clause.OnConflict{
+			Columns:   []clause.Column{{Name: "name"}, {Name: "artifact"}},
+			DoUpdates: clause.AssignmentColumns([]string{"updated_at", "command", "every", "next_runtime", "backfill", "owner", "is_active"}),
+		}).Create(&wf)
+	ctrl.db.Unscoped().Model(&wf).Update("deleted_at", nil)
 	c.JSON(http.StatusOK, "OK")
 }
 
+func (ctrl *Control) deleteWorkflow(c *gin.Context) {
+	var body deleteWorkflowReq
+	if err := c.BindJSON(&body); err != nil {
+		// bad request
+		c.JSON(http.StatusBadRequest, gin.H{"message": "could not parse body"})
+		return
+	}
+	var wf = table.Workflow{}
+	dbResult := ctrl.db.Where("name = ? and artifact = ?", body.Name, body.Artifact).First(&wf)
```

Comment:
you should be able to just delete, no? do you have to have an explicit select?

Reply:
I want to delete foreign key WorkflowSchedulerLock record (L99) if any, and that needs to find the workflows.ID.

Reply:
I will test if just delete will take care of the foreign key record in WorkflowSchedulerLock table.

Reply:
I think it's ok to keep the record in scheduler lock, it's an operational table anyway

Reply:
on the lock operation rules; the lock_time is not checked in the current code.  Is the plan to later have some elapse / expire check using the lock_time?

Reply:
> I was thinking a scenario, wf posted for creation. then sprinkler scheduler running, thus put in lock. Then a delete wf happens. Later, same wf name artifact posted again (thus wf deleted_at cleared). But the lock in place, and the scheduler would not act on it. Thus appearing as if there is a scheduler bug.

lock means it's still trying to schedule it right? it's supposed to be still locked until it finished

Reply:
yes, for the thread that has given the matching token, it is trying to schedule.  

for a wf that has been deleted, then posted again.  it would then be scheduled by a new token that does not match, and would not get scheduled.

Reply:
let me test and report on the deleted, re-posted scenario.

Reply:
`scheduler` takes care of creating and deleted the lock record with a unique token upon each tick.  

I have taken out the unneeded logic to check and clean up the workflow_scheduler_lock records upon workflows record delete.  

The control POST, then DELETE is tested successfully.  Checking the running scheduler, the given wf stops getting scheduled upon DELETE.  

When POST again on the same wf (by name, artifact), the scheduler picks up on it again.